### ISSUE TITLE
StrEnum `value` property type regression

### ISSUE THREAD

**Issue by Jonas1312** [microsoft/pyright#7983]
Hi,

This code used to work using pyright (python package) v1.1.360:

```python
from enum import StrEnum
from typing import Literal, assert_type


class Color(StrEnum):
    RED = "red"
    GREEN = "green"
    BLUE = "blue"


def foo(some_color: Color) -> None:
    assert_type(some_color.value, Literal["red", "green", "blue"])

```

```text
C:\Users\Jonas\Desktop\pyrightbug>pyright
WARNING: there is a new pyright version available (v1.1.360 -> v1.1.364).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`


added 1 package, and audited 2 packages in 7s

found 0 vulnerabilities
0 errors, 0 warnings, 0 informations 

```

With pyright v1.1.364 it returns this:

```text
c:\Users\Jonas\Desktop\pyrightbug\main.py
  c:\Users\Jonas\Desktop\pyrightbug\main.py:12:17 - error: "assert_type" mismatch: expected "Literal['red', 'green', 'blue']" but received "str" (reportAssertTypeFailure)
1 error, 0 warnings, 0 informations
```

Using python 3.12 on Windows.

I'm not sure if this is a regression or if this behavior is normal.

Thanks

**Comment by erictraut** [comment]
This change was related to [this issue](https://github.com/microsoft/pyright/issues/5092). Pyright's previous behavior was causing problems for the django library, which includes a custom metaclass that inherits from `EnumMeta`. 

I'll see if I can make the django case work without regressing the behavior of `StrEnum` and `IntEnum`.

**Comment by erictraut** [comment]
This is included in pyright 1.1.365.

**Comment by Jonas1312** [comment]
yep it works, thanks a lot
